openverse_catalog/docs/adding_a_new_provider.md

```diff
@@ -0,0 +1,90 @@
+# Openverse Providers
+
+## Overview
+
+The Openverse Catalog collects data from the APIs of sites that share openly-licensed media,and saves them in our Catalog database. This process is automated by Airflow DAGs generated for each provider. A simple provider DAG looks like this:
+
+![Example DAG](assets/provider_dags/simple_dag.png)
+
+At a high level the steps are:
+
+1. `generate_filename`: Generates a TSV filename used in later steps
+2. `pull_data`: Actually pulls records from the provider API, collects just the data we need, and commits it to local storage in TSVs.
+3. `load_data`: Loads the data from TSVs into the actual Catalog database, updating old records and discarding duplicates.
+4. `report_load_completion`: Reports a summary of added and updated records.
+
+When a provider supports multiple media types (for example, `audio` *and* `images`), the `pull` step consumes data of all types, but separate `load` steps are generated:
+
+![Example Multi-Media DAG](assets/provider_dags/multi_media_dag.png)
+
+## Adding a New Provider
+
+Adding a new provider to Openverse means adding a new provider DAG. Fortunately, our DAG factories automate most of this process. To generate a fully functioning provider DAG, you need to:
+
+1. Implement a `ProviderDataIngester`
+2. Add a `ProviderWorkflow` configuration class
+
+### Implementing a `ProviderDataIngester` class
+
+We call the code that pulls data from our provider APIs "Provider API scripts". You can find examples in [`provider_api_scripts` folder](../dags/providers/provider_api_scripts). This code will be run during the `pull` steps of the provider DAG.
+
+At a high level, a provider script should iteratively request batches of records from the provider API, extract data in the format required by Openverse, and commit it to local storage. Much of this logic is implemented in a [`ProviderDataIngester` base class](../dags/providers/provider_api_scripts/provider_data_ingester.py) (which also provides additional testing features *<TODO: link to documentation for testing features like ingestion_limit, skip_ingestion_errors etc>*). All you need to do to add a new provider is extend this class and implement its abstract methods.
```

Comment:
```suggestion
At a high level, a provider script should iteratively request batches of records from the provider API, extract data in the format required by Openverse, and commit it to local storage. Much of this logic is implemented in a [`ProviderDataIngester` base class](../dags/providers/provider_api_scripts/provider_data_ingester.py) (which also provides additional testing features *<TODO: link to documentation for testing features like ingestion_limit, skip_ingestion_errors etc>*). To add a new provider, extend this class and implement its abstract methods.
```

To remove any claims about simplicity. This is only one possible alternative.

Reply:
Good point 😅